Std/Data/List/Lemmas.lean

```diff
@@ -646,7 +646,7 @@ theorem head!_of_head? [Inhabited α] : ∀ {l : List α}, head? l = some a →
   | _a::_l, rfl => rfl
 
 theorem head?_eq_head : ∀ l h, @head? α l = some (head l h)
-  | [], h => nomatch h rfl
+  | [], h => nomatch h
   | _::_, _ => rfl
```

Comment:
If I'm not mistaken, the first case is not necessary:
```suggestion
  | _::_, _ => rfl
```

Reply:
I applied this suggestion.